web/libs/datamanager/src/utils/urlJSON.js

```diff
@@ -0,0 +1,20 @@
+// Library that serializes and deserializes JSON data to/from URL-safe strings
+// Uses gzip to minimize the size of the serialized data
+
+import { gzip, ungzip } from 'pako'
+import { fromUint8Array, toUint8Array } from 'js-base64'
+import { packJSON } from './packJSON';
+
+export const deserializeJsonFromUrl = (s) => {
+    // Fall back to packJSON if deserialization fails; this probably means
+    // we are deserializing an old virtual view. TODO(jo): deprecate packJSON entirely
```

Comment:
@jombooth, what's a path to deprecating all old virtual views?

Reply:
Whenever we're ready, we can cut a new version that removes deserialization via packJSON from the exception handler below, and at that point we can delete packJSON entirely. I've chatted with @amymrou about when we can do this, following the change management best practices that we're discussing - we haven't locked those yet, so for now, I've just cut a ticket for the deprecation work (LEAP-914) and whenever we pick that up, some lucky engineer can enjoy ripping ~600 lines of buggy code out of LS :)